src/components/Editor/index.js

```diff
@@ -568,14 +570,22 @@ const Editor = React.createClass({
       }));
   },
 
-  editorHeight() {
-    const { selectedSource, horizontal } = this.props;
+  getInlineEditorStyles() {
+    const { selectedSource, horizontal, searchOn } = this.props;
 
+    // The footer is off if there is no 'prettify' button (which happens for
+    // source-mapped or already prettified files)
     if (!shouldShowFooter(selectedSource, horizontal)) {
-      return "100%";
+      if (searchOn) {
+        return { height: "calc(100% - 54px)" };
+      }
+
+      return { height: "100%" };
     }
 
-    return "";
+    // The height is already set and so we dom't want to pass any values here.
+    // Set in editor.css (see .search-bar ~ .editor-mount)
+    return {};
```

Comment:
i think `.search-bar ~ .editor-mount` might be more confusing than it's worth now that we're doing inline styles. 

How about we move that value in here?